### PR TITLE
some name changes

### PR DESCRIPTION
**Pull request by millerjoey**
Changed some things to `snake_case` to match the MuttsDicts API. I'll be `add`ing this branch in my branch of our repo.

More meaningful changes:
- I switched from `copy` to `make_mutable_copy` for the sake of clarity so a user can write a method for that instead of overloading `Base.copy`.
- `is_mutts_mutable` instead of `ismutable` so there's no confusion between a mutable struct and a mutable `Mutts` object.
- I added `flag_immutable` (is there a better name for this?) so a user can optionally use some other mutable flag than that as created with the `@mutt` macro.

### CODE REVIEW

File: examples/MuttsVersionedTrees/MuttsVersionedTrees.jl
Comment (millerjoey) [inline examples/MuttsVersionedTrees/MuttsVersionedTrees.jl]: Should I change to `mutable_version`, suggested in #10?

File: src/Mutts.jl
Comment (ghost) [inline src/Mutts.jl]: Mmm good call! Yeah, this is definitely better. Thanks!

File: src/Mutts.jl
Comment (ghost) [inline src/Mutts.jl]: Maybe a better name could be `set_immutable_flag!(obj)`?

File: src/Mutts.jl
Comment (ghost) [inline src/Mutts.jl]: My gut says it would be better to just leave this unimplemented by default, a la:
```suggestion
function make_mutable_copy end
```
This is also what Lyndon suggests here in his blog post:
https://www.oxinabox.net/2020/04/19/Julia-Antipatterns.html (under `NotImplemented Exceptions`), though actually I think this example is more okay than the one in his post, since you're not specifying type restrictions on the arguments.

the error message you currently have isn't very helpful because, e.g., you don't specify for _which_ type isn't implemented, and if you get this deep in a stack trace, it's not very helpful.

Comment (ghost) [reply src/Mutts.jl]: Still, I'm also sympathetic to wanting to provide a nice message like you've done here, so if you do want to keep it like this, maybe you can do it by still throwing a MethodError instead of `error()`?:
```suggestion
function make_mutable_copy(obj)
    @warn "Implement some version of `make_mutable_copy` for your type."
    throw(MethodError(make_mutable_copy, typeof(obj))
end
```
But that's a bit silly, so i'd probably just stick with not implementing this method at all 👍

File: examples/MuttsVersionedTrees/MuttsVersionedTrees.jl
Comment (ghost) [reply examples/MuttsVersionedTrees/MuttsVersionedTrees.jl]: 👍 Yeah, that makes sense! I'm still unsure about what is better. I wish there was a standard that we could defer to here.... but yeah, i'd say go ahead and switch it, given @comnik's stated preference.

**Review by ghost — COMMENTED** [review]
Thanks for the renaming, @millerjoey! :)

My thoughts:
- `markimmutable!` => `mark_immutable!` 👍 
- `copy` => `make_mutable_copy` 👍 
- `ismutable` => `is_mutable()`:
    - i'm worried that it will be easy to mix these up, and users might accidentally write `ismutable(x)` when they mean `is_mutable(x)` 😮 
    - Can we maybe consider a more different name? Maybe `mutable(x)`? or `mutts_mutable(x)`? Or `can_mutate(x)`? Or i dunno something like that? Do you have preference? 
    - When we first wrote the MuttsDict.jl PR, `Base.ismutable(x)` didn't exist; it was added in julia 1.5. It's annoying; I'm not sure what's best to do about it. We could make all of these functions `mutts_is_mutable(x)`, `mutts_mark_immutable!(x)`, etc, but that's also lame... Maybe we just leave it how you have it for now and see how it goes?
- `flag_immutable!(obj)` -- i like this change. I've left a comment on names

**Comment by ghost** [comment]
Also, feel free to add yourself to the authors list here, if you don't mind being one of the "maintainers" who might answer questions about this package and stuff in the future!:
https://github.com/RelationalAI-oss/Mutts.jl/blob/master/Project.toml

File: src/Mutts.jl
Comment (ghost) [inline src/Mutts.jl]:
```suggestion
Freeze `obj` from further mutations, making it eligible to pass to
other Tasks, branch! from it, or otherwise share it.
```
Hehe just noticed this typo

File: src/Mutts.jl
Comment (ghost) [inline src/Mutts.jl]:
```suggestion
@generated function mark_immutable!(::MuttsType, obj::T) where T
```
Also, these spaces were old style

File: src/Mutts.jl
Comment (millerjoey) [reply src/Mutts.jl]: How about `is_mutts_mutable`? We already have `ismuttstype` (which I will change to `is_mutts_type`)

File: src/Mutts.jl
Comment (millerjoey) [reply src/Mutts.jl]: Thanks. Yeah, your initial suggestion makes a lot of sense. Changing to `function make_mutable_copy end`

File: examples/MuttsVersionedTrees/MuttsVersionedTrees.jl
Comment (millerjoey) [reply examples/MuttsVersionedTrees/MuttsVersionedTrees.jl]: Done

File: src/Mutts.jl
Comment (millerjoey) [reply src/Mutts.jl]: Done

**Comment by millerjoey** [comment]
Note the changes:

`is_mutable` -> `is_mutts_mutable`
`ismuttstype` -> `is_mutts_type`.

Let me know if that works!

**Comment by ghost** [comment]
> `is_mutable` -> `is_mutts_mutable`

Yeah, this seems reasonable to me! Hopefully that's not too verbose? Might be nice to get @comnik's and @li1's opinions on whether you'd be okay with using `is_mutts_mutable(x)` throughout the codebase (MuttsDicts and the Pager), but it seems reasonable to me! :)

@comnik and @li1: my concern was that it might be too easy to mix up `is_mutable(x)` and `ismutable(x)`, and it would be bad to mix them up since they're both valid and both return booleans, but with very different meanings, so it's probably good to make the distinction clear.

This sounds great to me! :)

File: examples/MuttsVersionedTrees/src/MuttsVersionedTrees.jl
Comment (ghost) [inline examples/MuttsVersionedTrees/src/MuttsVersionedTrees.jl]:
```suggestion
```
This is not needed; you can already add methods to a function just by using the fully-qualified name, and I think this is what is preferred. (Although it doesn't seem to be specified in our style guide! 😮 https://github.com/RelationalAI/RAIStyle#module-imports)

File: examples/MuttsVersionedTrees/MuttsVersionedTrees.jl
Comment (ghost) [inline examples/MuttsVersionedTrees/MuttsVersionedTrees.jl]: Oops, this whole file looks like a mistake. If you don't mind, i'm going to just push a commit to this branch to delete this file.

File: src/Mutts.jl
Comment (ghost) [inline src/Mutts.jl]:
```suggestion
    return make_mutable_copy(obj)
```

File: test/Mutts.jl
Comment (ghost) [inline test/Mutts.jl]: Same here
```suggestion
```

**Review by ghost — APPROVED** [review]
👍 Awesome. Thanks @millerjoey! :)

**Comment by ghost** [comment]
I've left a few more suggestions, but once those are addressed, feel free to merge! :)

**Comment by ghost** [comment]
And then please do start the Julia Pkg registration! :)

File: examples/MuttsVersionedTrees/src/MuttsVersionedTrees.jl
Comment (millerjoey) [reply examples/MuttsVersionedTrees/src/MuttsVersionedTrees.jl]: Oh, TIL. Thanks for noticing that.

File: examples/MuttsVersionedTrees/MuttsVersionedTrees.jl
Comment (millerjoey) [reply examples/MuttsVersionedTrees/MuttsVersionedTrees.jl]: Haha yes I should have asked about that!

**Comment by ghost** [comment]
Hmmmmm also, before you merge, it appears that Travis isn't running the tests? is that right?

**Comment by millerjoey** [comment]
Oh yes, I don't know why that is. I tested on my computer and they should pass. Any ideas?

**Review by ghost — APPROVED** [review]
Hooray, tests pass! 🙌 🤝 🚀  
LGTM ✅

**Pull request merged**
